includes/admin/reports/class-wc-report-downloads.php

```diff
@@ -0,0 +1,251 @@
+ * @author      WooThemes
+ * @category    Admin
+ * @package     WooCommerce/Admin/Reports
+ * @version     3.3.0
+ */
+class WC_Report_Downloads extends WP_List_Table {
+
+	/**
+	 * Max items.
+	 *
+	 * @var int
+	 */
+	protected $max_items;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+
+		parent::__construct( array(
+			'singular'  => 'download',
+			'plural'    => 'downloads',
+			'ajax'      => false,
+		) );
+	}
+
+	/**
+	 * Don't need this.
+	 *
+	 * @param string $position
+	 */
+	public function display_tablenav( $position ) {
+
+		if ( 'top' !== $position ) {
+			parent::display_tablenav( $position );
+		}
+	}
+
+	/**
+	 * Output the report.
+	 */
+	public function output_report() {
+
+		$this->prepare_items();
+
+		// Subtitle for permission if set
+		if ( isset( $_GET['permission_id'] ) && is_numeric( $_GET['permission_id'] ) && $_GET['permission_id'] > 0 ) {
+			// Load the permission, order, etc. so we can render more information
+			$permission	= null;
+			$product	= null;
+			try {
+				$permission	= new WC_Customer_Download( $_GET['permission_id'] );
+				$product	= wc_get_product( $permission->product_id );
+			} catch ( Exception $e ) {
+				echo '<p>Permission #' . esc_html( $_GET['permission_id'] ) . ' not found.</p>';
+			}
+
+			if ( ! empty( $permission ) && ! empty( $product ) ) {
+				echo '<h2>';
+
+				// File information
+				if ( $file = $product->get_file( $permission->get_download_id() ) ) {
+					echo esc_html( $file->get_name() ) . ' for ';
+				}
+
+				// Product information
+				$product_description = '';
+				if ( $sku = $product->get_sku() ) {
+					$product_description .= $sku . ' - ';
+				}
+				$product_description .= $product->get_name();
+				echo edit_post_link( $product_description, '', '', $product->get_id() );
+
+				// Order information
+				echo ' in Order ';
+				echo edit_post_link( '#' . $permission->order_id, '', '', $permission->order_id );
+
+				echo '</h2>';
+			}
+		}
+
+		echo '<div id="poststuff" class="woocommerce-reports-wide">';
+		$this->display();
+		echo '</div>';
+	}
+
+	/**
+	 * Get column value.
+	 *
+	 * @param mixed $item
+	 * @param string $column_name
+	 */
+	public function column_default( $item, $column_name ) {
+
+		$permission = null;
+		$product = null;
+		try {
+			$permission	= new WC_Customer_Download( $item->permission_id );
+			$product	= wc_get_product( $permission->product_id );
+		} catch ( Exception $e ) {
+			// Ok to continue rendering other information even if permission and/or product is not found
+		}
+
+		switch ( $column_name ) {
+			case 'timestamp' :
+				echo esc_html( $item->timestamp );
+				break;
+
+			case 'product' :
+				if ( empty( $product ) ) {
+					break;
+				}
+
+				$product_description = '';
+				if ( $sku = $product->get_sku() ) {
+					$product_description .= $sku . ' - ';
+				}
+				$product_description .= $product->get_name();
+
+				echo edit_post_link( $product_description, '', '', $product->get_id() );
```

Comment:
edit_post_link does not need an echo it seems. This error is repeated throughout the PR and needs correction.